### ISSUE TITLE
Handle the function with struct as a return type

### ISSUE THREAD

**Issue by Thirumalai-Shaktivel** [lfortran/lfortran#1593]
Example:
```fortran
module module_01
    implicit none
    type t
        integer :: i
    end type t
contains
    function new_value() result(p)
        type(t) :: p
        p%i = 123
    end function new_value
end module module_01

program derived_types_13
    use module_01
    implicit none

    type(t) :: x
    x = new_value()
    print *, x%i
end program derived_types_13
```

We remove the struct as a return type in subroutine_from_function pass. This cause the following asr_verfiy error:
```
ASR verify pass error: ASR verify: FunctionCall::m_name must be returning a non-void value.
```

I think we have to replace this return_var with a zero.

See: https://github.com/lfortran/lfortran/pull/1485#discussion_r1174406527

**Comment by certik** [comment]
Yes, we need to fix this. Not sure about replacing return_var with a zero, but we'll find a good fix.

**Comment by Thirumalai-Shaktivel** [comment]
The problem was in the `subroutine_from_function` pass; it seems the `FunctionCall` was not replaced with the `SubroutineCall`, so the error.

This is fixed here: 930d8df0c4d9a2e372bd5bf80f81b84acd1efc06

**Comment by Thirumalai-Shaktivel** [comment]
This issue is fixed.
```console
$ lfortran a.f90                    
123
```
